src/firstboot/firstboot.c

```diff
@@ -1161,6 +1163,76 @@ static int process_root_account(int rfd) {
 }
 
+static int process_regular_accounts(int rfd) {
+        _cleanup_close_ int creds_dfd = -EBADF, dest_dfd = -EBADF;
+        _cleanup_closedir_ DIR *creds_dir = NULL;
+        int r;
+
+        creds_dfd = open_credentials_dir();
+        if (creds_dfd < 0)
+                return log_error_errno(creds_dfd, "Failed to open credentials dir: %m");
+
+        creds_dir = take_fdopendir(&creds_dfd);
+        if (!creds_dir)
+                return log_error_errno(errno, "Failed to reopen credentials dir: %m");
+
+        FOREACH_DIRENT(de, creds_dir, return log_error_errno(errno, "Failed to read credentials dir: %m")) {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *identity = NULL;
+                _cleanup_close_ int dest_fd = -EBADF;
+                _cleanup_fclose_ FILE *dest = NULL;
+                _cleanup_free_ char *fn = NULL;
+                char *username;
+
+                username = startswith(de->d_name, "user.create.");
+                if (!username)
+                        continue;
+
+                if (!valid_user_group_name(username, 0)) {
+                        log_notice("Skipping over credential with name that is not a suitable user name: %s", de->d_name);
+                        continue;
+                }
+
+                 r = sd_json_parse_file_at(
+                                /* f= */ NULL,
+                                dirfd(creds_dir),
+                                de->d_name,
+                                /* flags= */ 0,
+                                &identity,
+                                /* ret_line= */ NULL,
+                                /* ret_column= */ NULL);
+                if (r < 0) {
+                        log_warning_errno(r, "Failed to parse user record in credential '%s', ignoring: %m", de->d_name);
+                        continue;
+                }
+
+                fn = strjoin(username, ".user");
+                if (!fn)
+                        return log_oom();
+
+                if (dest_dfd < 0) {
+                        dest_dfd = open_mkdir_at(rfd, "/etc/userdb/", O_CLOEXEC, 0755);
+                        if (dest_dfd < 0)
+                                return log_error_errno(dest_dfd, "Failed to create userdb drop-in dir: %m");
+                }
+
+                dest_fd = openat(dest_dfd, fn, O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0600);
+                if (dest_fd < 0)
+                        return log_error_errno(errno, "Failed to create .identity file in home directory: %m");
+
+                dest = take_fdopen(&dest_fd, "w");
+                if (!dest)
+                        return log_oom();
+
+                r = sd_json_variant_dump(identity, SD_JSON_FORMAT_PRETTY|SD_JSON_FORMAT_FLUSH, dest, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to write /etc/userdb/%s: %m", fn);
+
+                log_info("Created %s from %s credential", fn, de->d_name);
```

Comment:
this needs to be a bit more complex, see the relevant part from nss-systemd. Specifically:

1. you need to also create a symlink named after the numeric uid, pointing to the user record file. (so that resolution works both ways)
2. you need to strip the "privileged" part from the user record when writing it to the user record file, and then write it instead info a file `<username>.user-privileged`. There are ready made helpers for stripping/extracting these sections, so this should be easy.
3. We need some handling for group files too i guess.
4. We need to iterate through all groups a user record is part of, and all users listed as members of any groups and create .membership files for them. This is currently not documented, which is a bad omission. I'll prep a fix for that.